models/exposure.py

```diff
@@ -363,8 +363,25 @@ def __str__(self):
 
     def save(self):
         pass  # TODO: implement this! do we need this?
+        # Considertions for whether we need this:
+        # IF we believe that the place we got the exposure from will
+        #  have it in perpetuity, then we don't need to save it to
+        #  either our own database or our own archive.  (Or, if we trust
+        #  that once we've extracted images from it and saved those,
+        #  we'll never have to exract images again, then we don't need
+        #  to save the exposure.  This is more dubious, as some things
+        #  you do to images, e.g. sky subtraction, can be potentially
+        #  destructive.)  We can justpull it back from the source if we
+        #  ever need it again.
+        # But, if we think we need to save it to our own archive,
+        #  we will need this.
+        # It might be sufficient to just use FileOnDiskMixin.save().
```

Comment:
By download here, what I really mean is download to the local filesystem for processing.

So, there are two models: exposures are on the external source, and we pull them as necessary from that external source, trusting that they will never disappear.  In this case, we never need to save them to our own archive (only to the local filesystem where we're doing the work.)

In the other model, exposures are on the external source, but also we keep a copy of them in our own datastore.  In that case, we need to implement the full save method for the exposure, and exposures need to be saved to the archive.

Depending on which we choose, there are further implications.  If we aren't going to save exposures to our own archive, then we're going to have exposure objects with null md5sum fields (since md5sum is part of FileOnDiskMixin), so that needs to be allowed.  If we do want to save exposures to our own archive, we have to implmeent the save method (which might be just a call to super().save()).

I'm mixed about this.  Saving exposures to our own archive is gratuitous extra disk space and network time used up. But, I don't know if we want to trust that the external source will have them in perpetuity.

Reply:
Yes, this is a design decision. I am leaning towards having the data on our own controlled archive. That comes with all sorts of assumptions e.g., on the existence of a DB row only allowed when the data is verified on archive, and so on. 
Let's discuss this next time we have a call.